packages/bun-types/globals.d.ts

```diff
@@ -3876,3 +3876,30 @@ interface Navigator {
 }
 
 declare var navigator: Navigator;
+
+// -----------------------
+// -----------------------
+// --- ts builtins (lib.es5)
+
+interface JSON {
+  /**
+   * Converts a JavaScript Object Notation (JSON) string into an object.
+   * @param text A valid JSON string.
+   * @param reviver A function that transforms the results. This function is called for each member of the object.
+   * If a member contains nested objects, the nested objects are transformed before the parent object is.
+  */
+  parse(text: string, reviver?: (this: void, key: string, value: any) => any): unknown;
+}
+
+// Allow `<Array>.filter(Boolean)` to properly reflect
+interface Array<T> {
+  filter(predicate: BooleanConstructor, thisArg?: any): T[];
```

Comment:
oh i didnt realize what this is doing. @colinhacks 👀 what do you think about adding this override?